layerindex/tools/import_otherdistro.py

```diff
@@ -0,0 +1,789 @@
+            existing = list(layerrecipes.filter(deleted=False).values_list('pn', flat=True))
+
+            def handle_pkg(pkg):
+                pkgname = pkg['Package']
+                recipe, created = ClassicRecipe.objects.get_or_create(layerbranch=layerbranch, pn=pkgname)
+                if created:
+                    logger.info('Importing %s' % pkgname)
+                elif recipe.deleted:
+                    logger.info('Restoring and updating %s' % pkgname)
+                    recipe.deleted = False
+                else:
+                    logger.info('Updating %s' % pkgname)
+                filename = pkg.get('Filename', '')
+                if filename:
+                    recipe.filename = os.path.basename(filename)
+                    recipe.filepath = os.path.dirname(filename)
+                recipe.section = pkg.get('Section', '')
+                description = pkg.get('Description', '')
+                if description:
+                    description = description.splitlines()
+                    recipe.summary = description.pop(0)
+                    recipe.description = ' '.join(description)
+                recipe.pv = pkg.get('Version', '')
+                recipe.homepage = pkg.get('Homepage', '')
+                recipe.license = pkg.get('License', '')
+                recipe.save()
+                if pkgname in existing:
+                    existing.remove(pkgname)
+                if updateobj:
+                    rupdate, _ = ComparisonRecipeUpdate.objects.get_or_create(update=updateobj, recipe=recipe)
+                    rupdate.meta_updated = True
+                    rupdate.save()
+
+            pkgs = []
+            pkginfo = {}
+            lastfield = ''
+            with open(args.pkglistfile, 'r') as f:
+                for line in f:
+                    linesplit = line.split()
+                    if line.startswith('Package:'):
+                        # Next package starting, deal with the last one (unless this is the first)
+                        if pkginfo:
+                            handle_pkg(pkginfo)
+
+                        pkginfo = {}
+                        lastfield = 'Package'
+
+                    if line.startswith(' '):
+                        if lastfield:
+                            pkginfo[lastfield] += '\n' + line.strip()
+                    elif ':' in line:
+                        field, value = line.split(':', 1)
+                        pkginfo[field] = value.strip()
+                        lastfield = field
+                    else:
+                        lastfield = ''
+                if pkginfo:
+                    # Handle last package
+                    handle_pkg(pkginfo)
+
+                if existing:
+                    logger.info('Marking as deleted: %s' % ', '.join(existing))
+                    layerrecipes.filter(pn__in=existing).update(deleted=True)
+
+                layerbranch.vcs_last_fetch = datetime.now()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC. The recommended way to create an aware datetime object representing a specific timestamp in UTC is by passing `tzinfo` as an argument to the method.

[Learn more](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects)